linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_pam_unix_rounds_system_auth/ansible/shared.yml

```diff
@@ -3,163 +3,6 @@
-        replace: '\g<1> \g<2> rounds={{ var_password_pam_unix_rounds }}'
-      when:
-        - result_authselect_profile is not skipped
-        - result_pam_unix_rounds_present.found == 0
-
-    - name: Ensure a backup of current authselect profile before selecting the custom profile
-      ansible.builtin.command:
-        cmd: authselect apply-changes -b --backup=before-rounds-hardening.backup
-      when:
-        - result_authselect_check_cmd is success
-        - result_authselect_profile is not skipped
-        - authselect_current_profile is not match("custom/")
-        - authselect_custom_profile is not match(authselect_current_profile)
-
-    - name: Ensure the custom profile is selected
-      ansible.builtin.command:
-        cmd: authselect select {{ authselect_custom_profile }} --force
-      register: result_pam_authselect_select_profile
-      when:
-        - result_authselect_check_cmd is success
-        - result_authselect_profile is not skipped
-        - authselect_current_profile is not match("custom/")
-        - authselect_custom_profile is not match(authselect_current_profile)
-
-    - name: Restore the authselect features in the custom profile
-      ansible.builtin.command:
-        cmd: authselect enable-feature {{ item }}
-      loop: "{{ result_authselect_features.stdout_lines }}"
-      when:
-        - result_authselect_profile is not skipped
-        - result_authselect_features is not skipped
-        - result_pam_authselect_select_profile is not skipped
-
-    - name: Ensure the custom profile changes are applied
-      ansible.builtin.command:
-        cmd: authselect apply-changes -b --backup=after-rounds-hardening.backup
-      when:
-        - result_authselect_check_cmd is success
-        - result_authselect_profile is not skipped
-  when:
-  - result_authselect_present.stat.exists
-
-# For systems without authselect
-- name: "Remediation where authselect tool is not present and PAM files are directly edited"
-  block:
-    - name: Ensure the desired rounds value is updated in the custom profile
-      ansible.builtin.replace:
-        dest: "/etc/pam.d/system-auth"
-        regexp: '(^\s*password.*pam_unix.so.*rounds=)(\S+)(.*)$'
-        replace: '\g<1>{{ var_password_pam_unix_rounds }}\g<3>'
-
-    - name: Ensure the remember parameter is included in the custom profile
-      ansible.builtin.replace:
-        dest: "/etc/pam.d/system-auth"
-        regexp: '(^\s*password.*pam_unix.so.*)(?! rounds=\S+)(.*)$'
-        replace: '\g<1> \g<2> rounds={{ var_password_pam_unix_rounds }}'
-      when:
-        - result_pam_unix_rounds_present.found == 0
-  when:
-    - not result_authselect_present.stat.exists
+{{{ ansible_instantiate_variables("var_password_pam_unix_rounds") }}}
+{{{ ansible_ensure_pam_module_configuration('/etc/pam.d/system-auth', 'password', 'sufficient', 'pam_unix.so', 'rounds', '{{ var_password_pam_unix_rounds }}', '') }}}
```

Comment:
I don't think so. `pam_unix.so` is a critical module. If the control for it is different, a good reason should be provided for each case. Otherwise, it is safe to ensure the `sufficient` control. If necessary, exceptions can be created based on product.